cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/CustomActionExecutor.java

```diff
@@ -64,10 +67,12 @@ private WorkflowAction createAction(BasicWorkflowContext context, InstantiatorFa
     Class<?> clz = Class.forName(context.getSpecification().getClassName(), true, classLoader);
     Preconditions.checkArgument(WorkflowAction.class.isAssignableFrom(clz), "%s is not a WorkflowAction.", clz);
     WorkflowAction action = instantiator.get(TypeToken.of((Class<? extends WorkflowAction>) clz)).create();
+    Metrics metrics = new ProgramUserMetrics(
+      context.getProgramMetrics().childContext(Constants.Metrics.Tag.NODE, action.getClass().getSimpleName()));
```

Comment:
nit: use temporary variable for `context.getProgramMetrics` to fit into single line?